monicar-event-hub/src/main/java/org/eventhub/application/AlarmService.java

```diff
@@ -0,0 +1,56 @@
+package org.eventhub.application;
+
+import java.util.Optional;
+
+import org.eventhub.application.port.AlarmRepository;
+import org.eventhub.application.port.AlarmSender;
+import org.eventhub.domain.Alarm;
+import org.eventhub.infrastructure.external.command.AlarmSend;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Service
+@Slf4j
+public class AlarmService {
+	private final AlarmRepository alarmRepository;
+	private final AlarmSender alarmSender;
+	@Value("${alarm-interval-distance}")
+	private Integer alarmIntervalDistance;
+
+	public Optional<Alarm> findById(Long vehicleId) {
+		return alarmRepository.findByVehicleId(vehicleId);
+	}
+
+	@Transactional(propagation = Propagation.REQUIRES_NEW)
+	public Optional<Long> saveAlarmIfNecessary(Long vehicleId, Long totalDistance) {
+		boolean isNecessary = findById(vehicleId)
+			.map(alarm -> checkBiggerThanIntervalDistance(totalDistance, alarm.getDrivingDistance()))
+			.orElse(true);
+
+		Optional<Long> alarmId = Optional.empty();
+		if (isNecessary) {
+			alarmId = Optional.ofNullable(alarmRepository.save(vehicleId));
+		}
+		return alarmId;
```

Comment:
```suggestion
		if (isNecessary) {
			return Optional.ofNullable(alarmRepository.save(vehicleId));
		}
		return Optional.empty();
```
이렇게 작성하는 게 더 가독성이 좋지 않을까요??

Reply:
좋은데요??? 반영했습니다